cpp/ps0004/problem 4 p301F Salwa معضلة الاسكي فاليو.cpp

```diff
@@ -4,20 +4,31 @@ using namespace std;
 
 int main()
 {
-cout<<"--------------------------------------------" <<endl;
-cout<<"Problem 0003"<< endl;
-
-    int num1;
+    // prompting user for number 1/1.5
+    float num;
     cout << "enter a number of ASCII value \n";
-    cin>> num1;
+    cin >> num;
+    int temp = num; // to compare with user input
 
-    if(num1>=32 &num1<=126 )
-{
-   cout <<"the character for  "<< num1 << " is "<<char (num1) << "\n";
-}
-else
-{
-    cout <<"Invalid character/s or zero value \n";
-}
+    // checking invalidaty of user's input 2/4
+	// printing the number and the opposite printable character 2/2
+    if(num == 0)
+    {
+        cout << "Invalid character/s or zero value \n";
+    }
+    else if(num != temp)
+    {
+        cout << "Invalid: a floating point number! \n";
+    }
+    else if(num >= 32 && num <= 126)
+    {
+       cout << "the character for  " << num << " is " << char (num) << "\n";
+    }
+    else
+    {
+        cout << "Invalid: number is out of range! \n";
+    }

```

Comment:
- [x] used `if` structure in a right way
- [ ] `if` structure take in consideration all invalid cases
- [ ] all expressions to check invalidity are correct
- [ ] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character